packages/sdk-react/src/components/approvalRequests/ApprovalRequestsTable/ApprovalRequestsTable.tsx

```diff
@@ -0,0 +1,289 @@
+import React, { useState } from 'react';
+
+import { components } from '@/api';
+import { ScopedCssBaselineContainerClassName } from '@/components/ContainerCssBaseline';
+import { useMoniteContext } from '@/core/context/MoniteContext';
+import { MoniteScopedProviders } from '@/core/context/MoniteScopedProviders';
+import { useCurrencies } from '@/core/hooks';
+import { useEntityUserByAuthToken } from '@/core/queries';
+import { useIsActionAllowed } from '@/core/queries/usePermissions';
+import { AccessRestriction } from '@/ui/accessRestriction';
+import { CounterpartCell } from '@/ui/CounterpartCell';
+import { LoadingPage } from '@/ui/loadingPage';
+import {
+  TablePagination,
+  useTablePaginationThemeDefaultPageSize,
+} from '@/ui/table/TablePagination';
+import { DateTimeFormatOptions } from '@/utils/DateTimeFormatOptions';
+import { ActionEnum } from '@/utils/types';
+import { t } from '@lingui/macro';
+import { useLingui } from '@lingui/react';
+import { Box, Stack } from '@mui/material';
+import { DataGrid, GridValueFormatterParams } from '@mui/x-data-grid';
+
+import { addDays, formatISO } from 'date-fns';
+
+import {
+  FILTER_TYPE_STATUS,
+  FILTER_TYPE_CREATED_AT,
+  FILTER_TYPE_ADDED_BY,
+  FILTER_TYPE_CURRENT_USER,
+} from '../consts';
+import { FilterTypes, FilterValue } from '../types';
+import { ApprovalRequestsFilter } from './ApprovalRequestsFilter';
+import { ApprovalRequestStatusChip } from './ApprovalRequestStatusChip';
+import { ApproveButton } from './ApproveButton';
+import { RejectButton } from './RejectButton';
+import { UserCell } from './UserCell';
+
+interface ApprovalRequestsTableProps {
+  /**
+   * The event handler for a row click.
+   *
+   * @param id - The identifier of the clicked row, a string.
+   */
+  onRowClick?: (id: string) => void;
```

Comment:
@saladNights, we have to add such [callback](https://github.com/team-monite/monite-sdk/blob/94ad73e35649255c7a07152e901a6417d21afad9/packages/sdk-react/src/components/receivables/InvoicesTable/useInvoiceRowActionMenuCell.tsx#L25-L28) to `ApprovalRequestsTableProps`:
```ts
  onRowActionClick: (props: {
    id: string;
    action: InvoicesTableRowAction;
  }) => void;
```
and then we have to pass those callbacks inside of buttons:

```
<ApproveButton onClick={onRowActionClick} />
<RejectButton onClick={onRowActionClick} />
```

All mutations should be done in Super Component through similar callbacks.

Reply:
mutations were moved, callback was added